python/cudf/cudf/_libxx/string_casting.pyx

```diff
@@ -0,0 +1,545 @@
+    cdef data_type c_out_type = data_type(tid)
+    with nogil:
+        c_result = move(
+            cpp_to_integers(
+                input_column_view,
+                c_out_type))
+
+    return Column.from_unique_ptr(move(c_result))
+
+
+def i8tos(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type int8 to string column
+
+    Parameters:
+    -----------
+    input_col : input column of type int8
+
+    Returns
+    -------
+    A Column with int8 values cast to string
+    """
+
+    return integer_to_string(input_col)
+
+
+def stoi8(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type string to int8
+
+    Parameters:
+    -----------
+    input_col : input column of type string
+
+    Returns
+    -------
+    A Column with strings cast to int8
+    """
+
+    return string_to_integer(input_col, np.dtype("int8"))
+
+
+def i16tos(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type int16 to string column
+
+    Parameters:
+    -----------
+    input_col : input column of type int16
+
+    Returns
+    -------
+    A Column with int16 values cast to string
+    """
+
+    return integer_to_string(input_col)
+
+
+def stoi16(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type string to int16
+
+    Parameters:
+    -----------
+    input_col : input column of type string
+
+    Returns
+    -------
+    A Column with strings cast to int16
+    """
+
+    return string_to_integer(input_col, np.dtype("int16"))
+
+
+def itos(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type int32 to string column
+
+    Parameters:
+    -----------
+    input_col : input column of type int32
+
+    Returns
+    -------
+    A Column with int32 values cast to string
+    """
+
+    return integer_to_string(input_col)
+
+
+def stoi(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type string to int32
+
+    Parameters:
+    -----------
+    input_col : input column of type string
+
+    Returns
+    -------
+    A Column with strings cast to int32
+    """
+
+    return string_to_integer(input_col, np.dtype("int32"))
+
+
+def ltos(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type int64 to string column
+
+    Parameters:
+    -----------
+    input_col : input column of type int64
+
+    Returns
+    -------
+    A Column with int64 values cast to string
+    """
+
+    return integer_to_string(input_col)
+
+
+def stol(Column input_col, **kwargs):
+    """
+    Converting/Casting input column of type string to int64
+
+    Parameters:
+    -----------
+    input_col : input column of type string
+
+    Returns
+    -------
+    A Column with strings cast to int64
+    """
+
+    return string_to_integer(input_col, np.dtype("int64"))
+
+
+def _to_booleans(Column input_col, Scalar str_true=Scalar("True")):
```

Comment:
Is there a way we could make the default parameter here `None` or some other object that doesn't allocate GPU memory? Otherwise we allocate memory on import instead of at runtime which feels funky.

Reply:
I can, shouldn't be a problem

Reply:
@kkraus14 Do you think it's failing because we haven't don't have any resource yet ?
like rmm has not yet acquired memory

Reply:
It's failing because it's running on a machine without GPUs 😄

Reply:
Oh, only test CI runs on GOU machines and rest are not ? Never knew that :)